homeassistant/components/comfoconnect/config_flow.py

```diff
@@ -0,0 +1,158 @@
+        errors = {}
+        bridges = await self.hass.async_add_executor_job(Bridge.discover)
+        if bridges:
+            # one or more bridges auto-discovered
+            host_field_val = vol.All(str, vol.In([bridge.host for bridge in bridges]))
+            host_field_key = vol.Required(CONF_HOST, default=bridges[0].host)
+        else:
+            # no bridge discovered
+            host_field_val = str
+            host_field_key = vol.Required(CONF_HOST)
+        schema = vol.Schema(
+            {
+                host_field_key: host_field_val,
+                vol.Optional(CONF_NAME, default=DEFAULT_NAME): str,
+                vol.Optional(CONF_PIN, default=DEFAULT_PIN): int,
+            }
+        )
+        if user_input is not None:
+            unique_id = None
+            try:
+                unique_id = await validate_input(self.hass, user_input)
+            except ConnectionError:
+                errors[CONF_HOST] = "cannot_connect"
+            except PyComfoConnectNotAllowed:
+                errors[CONF_PIN] = "invalid_auth"
+            # We have to catch a very broad "Exception" here, since that's what
+            # pycomfoconnect raises
+            except Exception:  # pylint: disable=broad-except
+                errors["base"] = "unknown"
+            if unique_id:
+                await self.async_set_unique_id(unique_id)
+                self._abort_if_unique_id_configured()
+                user_input[CONF_TOKEN] = secrets.token_hex(16)
+                user_input[CONF_USER_AGENT] = DEFAULT_USER_AGENT
+                self.context[STEP_ID_USER] = user_input
+                return await self.async_step_sensor_selection()
+        return self.async_show_form(
+            step_id=STEP_ID_USER, data_schema=schema, errors=errors
+        )
+
+    async def async_step_sensor_selection(self, user_input=None):
+        """Handle sensor selection step."""
+        schema = vol.Schema(
+            {
+                vol.Optional(
+                    CONF_SENSORS,
+                ): cv.multi_select({sensor.key: sensor.name for sensor in SENSOR_TYPES})
+            }
+        )
+        errors = {}
+        unique_id = self.context["unique_id"]
+        self._abort_if_unique_id_configured()
+        if self.context["source"] == config_entries.SOURCE_IMPORT:
+            user_input = {"sensors": self.context["import_sensors"] or []}
+        if user_input is not None:
+            data = self.context[STEP_ID_USER]
+            return self.async_create_entry(
+                title=f"ComfoAir {unique_id}",
+                data=data,
+                options={"sensors": user_input["sensors"]},
+            )
+        return self.async_show_form(
+            step_id="sensor_selection",
+            data_schema=schema,
+            errors=errors,
+        )
+
+    async def async_step_import(self, import_config):
+        """Handle a flow import."""
+        return await self.async_step_user(import_config)
```

Comment:
Considering the defaults done in the validation, I think the data needs to be pre-processed here.